logistics/scripts/modules/settings/settings-management.psm1

```diff
@@ -312,21 +321,75 @@ function New-DevelopmentAppSettings([hashtable] $Settings = @{ }) {
 
         $newDevelopmentSettings = Add-TestHarnessSpecificAppSettings $newDevelopmentSettings $project
 
+        $newDevelopmentSettings = Merge-Hashtables $newDevelopmentSettings, (Get-CredentialSettingsByProject)[$project], $Settings
+
         $projectPath = Get-RepositoryResolvedPath $Project
 
         $newDevelopmentSettingsPath = Join-Path $projectPath "appsettings.development.json"
         New-JsonFile $newDevelopmentSettingsPath $newDevelopmentSettings -Overwrite
         $newSettingsFiles += $newDevelopmentSettingsPath
 
         $credentialSettings = Merge-Hashtables @{ }, (Get-CredentialSettingsByProject)[$project]
-        $newUserSettingsPath = Join-Path $projectPath "appsettings.user.json"
-        New-JsonFile $newUserSettingsPath $credentialSettings
-        $newSettingsFiles += $newUserSettingsPath
+        $userSecrets += ($credentialSettings | ConvertTo-Json -Depth 10 | dotnet user-secrets set --project $projectPath --id (Get-UserSecretsIdByProject).$project)
     }
 
     return $newSettingsFiles
 }
+
+$RegExFilter = "\A(?<curr>[^:]+)(:(?<remain>(?<sub>[^:]+)(:.*)?))?\Z"
+
+function Get-FlattenHash {
+    param([string]$Key,$Value,$Obj)  
+
+    if ($Value) {
+        if ($key -match $RegExFilter) {
+            $Current = $matches.curr
+            if (-not ($Obj | gm $Current)) {
+                        if (-not $Obj.ContainsKey($Current)) { $Obj.$Current = @{ } }
+            }
+
+            if ($matches.remain) {
+                $Obj.$Current = Get-FlattenHash -Key $matches.remain -Value $Value -Obj ($Obj.$Current)
+            } else {
+                $Obj.$Current = $Value
+            }
+            return $Obj
+        }
+    }
+}
+
+function Get-UserSecrets() {
+    $developmentSettingsByProject = Get-DefaultDevelopmentSettingsByProject
+    $inputTable = @{}
+    foreach ($project in $developmentSettingsByProject.Keys){
+        $projectPath = Get-RepositoryResolvedPath $project
+        $userSecretList= dotnet user-secrets list --project $projectPath --id (Get-UserSecretsIdByProject).$project | Out-String
+
+        if($userSecretList -NotLike "*No secrets configured for this application*" -And ($userSecretList -ne $null) )
+        {
+            $lineLength= $userSecretList.Split("\r?\n|\r").length
+            $line= ($userSecretList -split "\r?\n|\r")
+            for ($linecnt=0; $linecnt -lt $lineLength; $linecnt++){
+               if ([string]::IsNullOrEmpty($line[$linecnt]) -ne 'True' )
+               {
+                  $line[$linecnt]= $line[$linecnt].Remove($line[$linecnt].LastIndexOf(' '),1)
+                  $lastindex= $line[$linecnt].LastIndexOf('=')
+                  $elementkey= $line[$linecnt].Substring(0,$lastindex-1)
+                  $elementvalue= $line[$linecnt].Substring($lastindex+1)
+                  $inputTable.Add($elementkey,$elementvalue)
+               }
+            }
+        }
```

Comment:
I thought you were using [ConvertFrom-StringData](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-stringdata?view=powershell-7) to do this?

Reply:
I changed to ConvertFrom-StringData  now . Thanks